packages/preview/machiatto/0.2.0/lib.typ

```diff
@@ -0,0 +1,333 @@
+        let chapter = upper(text(size: 0.68em, current.body))
+        if current.numbering != none {
+          if is-odd {
+            align(aln)[#chapter #h(gap) #i]
+          } else {
+            align(aln)[#i #h(gap) #chapter]
+          }
+        }
+      }
+    },
+  )
+
+  // Configure equation numbering.
+
+  set math.equation(numbering: "(1)")
+
+  {
+    // Configure heading numbering.
+    set heading(numbering: "1.")
+
+    // Start chapters on a new page.
+
+    show heading.where(level: 1): it => {
+      if chapter-pagebreak {
+        pagebreak(weak: true)
+      }
+      it
+    }
+    body
+  }
+
+  if bibliography != none {
+    pagebreak()
+    show std-bibliography: set text(0.85em)
+    // Use default paragraph properties for bibliography.
+
+    show std-bibliography: set par(leading: 0.65em, justify: false, linebreaks: auto)
+    bibliography
+  }
+}
+
+#let def(title: none, fill: color.rgb("E0F2F1"), body) = {
+  block(
+    width: 100%,
+    inset: 10pt,
+    fill: fill,
+    radius: 5pt,
+    stroke: 1pt,
+    [
+      #text(weight: "bold")[#underline(stroke: 1pt)[Definition: #title]]
+
+      #body
+    ],
+  )
+}
+
+#let todo(body) = {
+  block(
+    width: 100%,
+    inset: 10pt,
+    fill: yellow,
+    radius: 4pt,
+    [
+      #text(weight: "bold")[TODO]
+
+      #body
+    ],
+  )
+}
+
+#let note(body) = {
+  block(
+    width: 100%,
+    inset: 10pt,
+    fill: color.rgb("FFECB3"),
+    radius: 4pt,
+    stroke: 1pt,
+    [
+      #text(weight: "bold")[Note:]
+
+      #body
+    ],
+  )
+}
+
+#let frame(title: none, body, fill: none, line-numbers: true, line-number-color: gray) = {
+  let stroke = black + 1pt
+  let radius = 5pt
+
+  box(stroke: stroke, radius: radius)[
+    #if title != none {
+      block(
+        stroke: stroke,
+        fill: fill,
+        inset: 0.5em,
+        width: 100%,
+        below: 0em,
+        radius: (top-left: radius, top-right: radius),
+        align(center)[*#title*],
+      )
+    }
+    #block(
+      width: 100%,
+      inset: (x: 0em, y: 0.5em),
+      {
+        if line-numbers {
+          // Split the body into lines
+          let lines = body.text.split("\n")
+          // Create a table with line numbers and code
+
+          table(
+            columns: (auto, 1fr),
+            inset: (x: 0.4em, y: 0.3em),
+            stroke: (y: 0.5pt + rgb(220, 220, 220)),
+            ..lines.enumerate().map(((i, line)) => {
+              (
+                align(right)[#text(fill: line-number-color)[#(i + 1)]],
+                align(left)[#raw(lang: body.lang, line)]
+              )
+            }).flatten(),
+          )
+        } else {
+          body
+        }
+      },
+    )
+  ]
+}
+
+#let code-file(file-path, lang, fill, line-numbers: true, line-number-color: gray) = {
+  frame(
+    title: file-path,
+    fill: fill,
+    line-numbers: line-numbers,
+    line-number-color: line-number-color,
+    [#raw(read(file-path), lang: lang)],
```

Comment:
`read` is relative to the file in which it is called, not to where the value representing the path is created. You should accept a string with file contents and call `read` from the template instead of taking a file path as input.